mage_ai/orchestration/db/models/schedules.py

```diff
@@ -308,6 +309,38 @@ def in_progress_runs(
             PipelineRun.passed_sla.is_(False),
         ).all()
 
+    @safe_db_query
+    def retry_pipeline_run(
```

Comment:
I would put this method in `pipeline_scheduler.py` instead of this file.

Reply:
Also, this method has duplicate with this method: https://github.com/mage-ai/mage-ai/blob/master/mage_ai/api/resources/PipelineRunResource.py#L147-L183
We should try reducing the duplicate code.

Reply:
comments addressed in https://github.com/mage-ai/mage-ai/pull/2512